src/core/analysis/PowerPosteriorAnalysis.cpp

```diff
@@ -239,18 +242,73 @@ void PowerPosteriorAnalysis::runStone(size_t idx, size_t gen, double burnin_frac
+     * ceil( powers.size() / num_processes ) rather than floor( powers.size() / num_processes ). This will be the process
+     * that gets to print its status to the standard output. Note that process_active has a PID of 0, and does not always
+     * satisfy this condition.
+     */
+    std::vector<size_t> start(num_processes);
+    std::vector<size_t> end(num_processes);
+    std::vector<size_t> ceil_pids;
+    
+    for (size_t i = 0; i < num_processes; ++i)
+    {
+        // see PowerPosteriorAnalysis::runAll()
+        start[i] = floor( ( floor( i / double(processors_per_likelihood)) / (double(num_processes) / processors_per_likelihood) ) * powers.size() );
+        end[i]   = floor( ( ceil((i+1)/double(processors_per_likelihood)) / (double(num_processes) / processors_per_likelihood) ) * powers.size() );
+        if (end[i] - start[i] == ceil((double)powers.size() / num_processes))
+        {
+            ceil_pids.push_back(i);
+        }
+    }
     
-    // print output for users
-    if ( process_active ==true )
+    size_t pid_to_print = *std::min_element(ceil_pids.begin(), ceil_pids.end());
+    
+    if (pid == pid_to_print)
     {
-        std::cout << "Step ";
-        for (size_t d = size_t( ceil( log10( idx+1.1 ) ) ); d < digits; d++ )
+        if (num_processes == 1)
+        {
+            std::cout << "Step ";
+            for (size_t d = size_t( ceil( log10(idx + 1.1) ) ); d < digits; d++)
+            {
+                std::cout << " ";
+            }
+            std::cout << (idx + 1);
+        }
+        else
         {
-            std::cout << " ";
+            size_t step = idx + 1 - start[pid_to_print]; // make sure step counter starts from 1
+            size_t lower_bound = (step - 1) * num_processes + 1;
+            size_t upper_bound = std::min( step * num_processes, powers.size() );
+            size_t offset = *std::max_element(start.begin(), start.end());
+            if (lower_bound != upper_bound)
+            {
+                std::cout << "Steps ";
+                for (size_t d = size_t( ceil( log10(lower_bound + 0.1) ) ); d < size_t( ceil( log10(offset + 0.1) ) ); d++)
+                {
+                    std::cout << " ";
+                }
+                std::cout << lower_bound << "--" << upper_bound;
+                for (size_t d = size_t( ceil( log10(upper_bound + 0.1) ) ); d < digits; d++)
+                {
+                    std::cout << " ";
+                }
+            }
+            else
+            {
+                std::cout << "Step ";
+                for (size_t d = 0; d < size_t( ceil( log10(offset + 0.1) ) ) + 3; d++)
+                {
+                    std::cout << " ";
+                }
+                std::cout << lower_bound;
+            }
         }
-        std::cout << (idx+1) << " / " << powers.size();
-        std::cout << "\t\t";
+    
+        std::cout << " / " << powers.size() << "\t\t";
```

Comment:
The off-by-one error could be fixed with:
```suggestion
        std::cout << " / " << (powers.size() - 1) << "\t\t";
```

Does `powers.size()` also need a `- 1` appending where it is used above?

Reply:
On closer inspection, I see that 129 steps _are_ in fact being performed – I missed the file ending `_stone_0.pp`. 
Perhaps the off by one error is occurring where `powers` is defined?

Reply:
@ms609 That's right – this is not a problem with reporting the progress of the analysis, but with setting it up in the first place. What goes into the `cats` argument of the `powerPosterior()` argument is actually not the number of stones / powers, but the largest index using 0-based indexing. So specifying `cats=127` produces stones indexed 0 through 127, i.e., 128 stones in total.

I've always found this strange, because unlike C++, Rev uses 1-based indexing, and even if that were not the case, it would still make more sense to specify the actual number rather than the largest index. It would be trivial to fix this, but I'm wary of making the change unilaterally, because it would silently change the behavior of existing scripts, potentially including published ones. Feedback by @bredelings @mlandis @hoehna would be much appreciated.

Reply:
I hear the hesitation about changing behaviour (and potentially changing - slightly - existing scripts).  Such a change would, however, bring the behaviour of the software in line with the [documentation](https://revbayes.github.io/documentation/powerPosterior.html), which reads (for `cats`): 
> The number of categories if no powers are specified